spec/api_config_spec.rb

```diff
@@ -0,0 +1,44 @@
+require 'spec_helper'
+
+describe Quandl::ApiConfig do
+  context 'threading' do
+    context '.api_key' do
+      let(:threads) { [] }
+      let(:default_key) { (0...8).map { (65 + rand(26)).chr }.join }
+
+      it 'should allows for a default value' do
+        Quandl::ApiConfig.api_key = default_key
```

Comment:
So what this will do is that if you set `Quandl::ApiConfig.api_key` in the main thread the new threads will take that value by default since they are `nil`. However if that thread changes the value it will change it only for that thread and leave the default alone. Basically the first time its set its saved as the default.